### PR TITLE
nette 3.1 compatibility

### PR DESCRIPTION
**Pull request by vrestihnat**
Simple adjustments for nette 3.1 and latte 2.10.1 compatibility. Please publish changes.

### CODE REVIEW

**Comment by lulco** [comment]
@enumag please can you take a look to this? It fixes master and other PR will have green build too. Thanks!

File: composer.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
 	"require": {
 		"php": ">=7.1",
 		"kdyby/strict-objects": "^2.0",
-		"latte/latte": "^2.5.2 || ^3.0",
+		"latte/latte": "^2.5.2 <= 2.10.1",
```
Comment (enumag) [inline composer.json:22]: why?

File: .travis.yml [hunk 2]
```diff
@@ -28,9 +28,9 @@ matrix:
       env: COMPOSER_EXTRA_ARGS="--prefer-stable" CODING_STANDARD=1
   exclude:
     - php: 7.3
-      env: COMPOSER_EXTRA_ARGS="--prefer-stable"
+      env: COMPOSER_EXTRA_ARGS="--prefer-stable"
```
Comment (enumag) [inline .travis.yml:31]: ?

Comment (vrestihnat) [reply .travis.yml]: fixed

File: composer.json (hunk 1 above)
Comment (vrestihnat) [reply composer.json:22]: There is a compatibility issue with latte/latte:dev-master. There are BCs that are not compatible with nette/application. Specifically: Nette\Bridges\ApplicationDI\LatteExtension. There is probably a backlog from nette developers. I hope they fix it soon. I'm helpless at the moment.

Comment (enumag) [reply composer.json:22]: it seems the next latte version will be 3.0 though so this shouldn't be necessary here

Comment (vrestihnat) [reply composer.json]: I forgot about the --prefer-stable local test. Now there is the original dependence.

File: tests/KdybyTests/Translation/LatteExtractorTest.phpt [hunk 3]
```diff
@@ -23,17 +24,32 @@ class LatteExtractorTest extends \KdybyTests\Translation\TestCase
...
+					'Important title' => 'Important title',
+					'Another important title' => 'Another important title',
+					"\nInteresting article about interesting topic\n" => "\nInteresting article about interesting topic\n",
+					'Chapter 2' => 'Chapter 2',
+					'none|one|many' => 'none|one|many',
+					'sample.identificator' => 'sample.identificator',
+				],
+			];
+		} else {
+			// we like cs :-)
```
Comment (enumag) [inline tests/KdybyTests/Translation/LatteExtractorTest.phpt:39]:
```suggestion
```

File: tests/KdybyTests/Translation/LatteExtractorTest.phpt [hunk 4]
```diff
@@ -43,17 +58,34 @@ class LatteExtractorTest extends \KdybyTests\Translation\TestCase
...
+					'homepage.Chapter 2' => 'Chapter 2',
+					'homepage.none|one|many' => 'none|one|many',
+					'homepage.sample.identificator' => 'sample.identificator',
+				],
+			];
+		} else {
+			$impT = sprintf('%s%s%s', '(', '"Important title"', ')');
+			$prefixImpT = sprintf('%s%s%s', 'homepage.(', '"Important title"', ')');
+			$prefixNoneOneMany = sprintf('%s%s', 'homepage.', 'none|one|many');
+			$prefixSampleIdentificator = sprintf('%s%s', 'homepage.sample.', 'identificator');
```
Comment (enumag) [inline tests/KdybyTests/Translation/LatteExtractorTest.phpt:76]: These variables seem useless, the result is the same as in the 2.9 case.

Comment (vrestihnat) [reply tests/KdybyTests/Translation/LatteExtractorTest.phpt]: I agree. It is fixed.

**Comment by lulco** [comment]
any news here?

File: tests/KdybyTests/Translation/LatteExtractorTest.phpt [hunk 5]
```diff
@@ -43,17 +58,34 @@ class LatteExtractorTest extends \KdybyTests\Translation\TestCase
...
+			$mess = [
+				'messages' => [
+					$prefixImpT => $impT,
+					"homepage.('Another important title')" => "('Another important title')",
```
[the remark this answers is not in the document]
Comment (vrestihnat) [reply tests/KdybyTests/Translation/LatteExtractorTest.phpt]: The problem is that this "(\"Important title\")"  does not meet the defined CS. I hope the following commit will explain everything and the test will be a little cleaner.

[the remark this answers is not in the document]
Comment (enumag) [reply tests/KdybyTests/Translation/LatteExtractorTest.phpt]: Why don't you use single quotes?

[the remark this answers is not in the document]
Comment (vrestihnat) [reply tests/KdybyTests/Translation/LatteExtractorTest.phpt]: The reason is that the assertion test fails.

[the remark this answers is not in the document]
Comment (vrestihnat) [reply tests/KdybyTests/Translation/LatteExtractorTest.phpt]: I will try to replace the outer quotes

File: src/DI/TranslationExtension.php [hunk 6]
```diff
@@ -297,12 +298,12 @@ public function beforeCompile()
 				->addSetup('addFilter', ['translate', [$this->prefix('@helpers'), 'translateFilterAware']]);
 		};
 
-		$latteFactoryService = $builder->getByType(ILatteFactory::class);
+		$latteFactoryService = $builder->getByType(LatteFactory::class)?: $builder->getByType(ILatteFactory::class);
```
Comment (enumag) [inline src/DI/TranslationExtension.php:301]:
```suggestion
		$latteFactoryService = $builder->getByType(LatteFactory::class) ?: $builder->getByType(ILatteFactory::class);
```

**Comment by lulco** [comment]
@vrestihnat you are very close :) please don't give up

**Comment by vrestihnat** [comment]
I don't give up. I just don't see any more comments so we can close it.

**Review by enumag — APPROVED** [review]

**Comment by enumag** [comment]
Please check if we also need #182 and/or #179. Don't wanna make a half-baked release.

**Comment by vrestihnat** [comment]
#182 is useful, it solves compatibility with pair macros. #179 is obsolete, #180 is resolved. I will prepare a pull request that contains  #182 + test

**Comment by vrestihnat** [comment]
#177 is partially solved and the rest is #182

**Pull request merged**